hazelcast-client/src/main/java/com/hazelcast/client/config/ClientConfig.java

```diff
@@ -288,7 +289,13 @@ public NearCacheConfig getNearCacheConfig(String name) {
         NearCacheConfig nearCacheConfig = lookupByPattern(nearCacheConfigMap, name);
         if (nearCacheConfig == null) {
             nearCacheConfig = nearCacheConfigMap.get("default");
+            if (nearCacheConfig != null) {
+                // if there is a default config we have to clone it,
+                // otherwise you will modify the same instances via different Near Cache names
+                nearCacheConfig = new NearCacheConfig(nearCacheConfig);
```

Comment:
Peter noticed that it's dangerous to use the `default` configuration without cloning, since we would return the same instance for different names. This should fix that (good finding!).